packages/language-server/src/quick-fix.ts

```diff
@@ -12,78 +12,182 @@ import { computeQuickFixStableIdInfo } from "@ui5-language-assistant/xml-views-q
   );
-  return CodeAction.create(
-    QUICK_FIX_STABLE_ID_COMMAND_TITLE,
-    Command.create(
+
+  codeActions.push(
+    CodeAction.create(
       QUICK_FIX_STABLE_ID_COMMAND_TITLE,
-      QUICK_FIX_STABLE_ID_COMMAND,
-      opts.document.uri,
-      replaceRange,
-      quickFixStableIdInfo.newText
-    ),
-    CodeActionKind.QuickFix
+      Command.create(
+        QUICK_FIX_STABLE_ID_COMMAND_TITLE,
+        QUICK_FIX_STABLE_ID_COMMAND,
+        opts.document.uri,
+        opts.document.version,
+        replaceRange,
+        quickFixStableIdInfo[0].newText
+      ),
+      CodeActionKind.QuickFix
+    )
+  );
+
+  const quickFixFileStableIdCodeActions = computeCodeActionsForQuickFixFileStableId(
+    {
+      document: opts.document,
+      xmlDocument: opts.xmlDocument,
+      ui5Model: opts.ui5Model,
+    }
   );
+
+  codeActions = codeActions.concat(quickFixFileStableIdCodeActions);
+
+  return codeActions;
+}
+
+function computeCodeActionsForQuickFixFileStableId(opts: {
+  document: TextDocument;
+  xmlDocument: XMLDocument;
+  ui5Model: UI5SemanticModel;
+}): CodeAction[] {
+  const validators = {
+    document: [],
+    element: [validateNonStableId],
+    attribute: [],
+  };
+
+  const nonStableIdFileIssues = validateXMLView({
+    validators,
+    model: opts.ui5Model,
+    xmlView: opts.xmlDocument,
+  });
+
+  if (nonStableIdFileIssues.length === 1) {
+    return [];
+  }
+
+  const errorsOffset = map(nonStableIdFileIssues, (_) => _.offsetRange);
+  const nonStableIdFileIssuesInfo = computeQuickFixStableIdInfo(
+    opts.xmlDocument,
+    errorsOffset
+  );
+  const nonStableIdFileIssuesLSPInfo: QuickFixStableIdLSPInfo[] = map(
+    nonStableIdFileIssuesInfo,
+    (_) => {
+      return {
+        newText: _.newText,
+        replaceRange: offsetRangeToLSPRange(_.replaceRange, opts.document),
+      };
+    }
+  );
+
+  return [
+    CodeAction.create(
+      QUICK_FIX_FILE_STABLE_ID_COMMAND_TITLE,
+      Command.create(
+        QUICK_FIX_FILE_STABLE_ID_COMMAND_TITLE,
+        QUICK_FIX_FILE_STABLE_ID_COMMAND,
+        opts.document,
+        opts.document.uri,
+        opts.document.version,
+        nonStableIdFileIssuesLSPInfo
+      ),
+      CodeActionKind.QuickFix
+    ),
+  ];
 }
 
 export function executeQuickFixStableIdCommand(opts: {
-  textDocument: TextDocument;
+  documentUri: string;
+  documentVersion: number;
   quickFixReplaceRange: LSPRange;
   quickFixNewText: string;
 }): TextDocumentEdit[] {
   const documentEdit = [
     TextDocumentEdit.create(
-      { uri: opts.textDocument.uri, version: opts.textDocument.version },
+      { uri: opts.documentUri, version: opts.documentVersion },
       [TextEdit.replace(opts.quickFixReplaceRange, `${opts.quickFixNewText}`)]
     ),
   ];
 
   return documentEdit;
 }
+
+export function executeQuickFixFileStableIdCommand(opts: {
+  documentUri: string;
+  documentVersion: number;
+  nonStableIdIssues: QuickFixStableIdLSPInfo[];
+}): TextDocumentEdit[] {
```

Comment:
Do we need `TextDocumentEdit[]` **array** in the return type?
it seems to always be an array sized **1**?

Or is that to support some generic pattern by the caller?

Reply:
It's a genetic pattern by the caller